Survival-DEV/blog-backend
# Blog-35: User Authorization

### :dart:  What is been done in this PR:
1. [x] Implement RBAC (Role-based access control ) policy for auth
2. [x] Add anew column in userEnitty for user `Role` permissions
3. [x] Install `CASL` library 
 **Related to #35 **

**In progress**
- [ ] the current code is under validation 
- [ ] set the actions that the admin is willing to manage & attach it into the related endpoints
----
### :compass:   How to test it
Steps to test my PR:
1. `npm run ci`
2. make sure to set up your `env` file to have these params:
   <details>
      <summary>Expand to see the params</summary>
    
      > DATABASE_URL=<your db url#>
      > NODE_ENV=DEBUG  //for now
      > logDB=["query", "error"]
      > PORT=<your port#>
      > TRUSTED_IP=["127.0.0.1"]
      > WHITELIST=['http://localhost']
      > ALLOWED_METHODS='GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS'
      > JWT_VERIFICATION_TOKEN_SECRET='secreto2'
      > JWT_VERIFICATION_TOKEN_EXPIRATION_TIME='60s'
      > EMAIL_CONFIRMATION_URL='http://localhost:5000/api/confirm-email/'
      > SEND_GRID_ACCESS_KEY=`<your sendgird key>`
      > FROM_EMAIL=`your sendgrid email`
      > SENDGRID_TEMPLATE_ID=d-3285e7571ee64e8fafedb163996e5a55
      >CONTACT_EMAIL= `<some_mail for_swagger_configs>`
   </details>
4. `docker-compose up`   --to fire redis 
3. `npm run start:dev`   --in another terminal
